### PR TITLE
feat(PB-69): shuffle and sort by post type feature flags added

### PR DESCRIPTION
**Pull request by michellejanay**
# (PB-69): Feed order flags

- shuffle, sort by post type, and show/hide labels feature flags added

## Related Ticket/PR

- [Jira ticket](https://whotargetsme.atlassian.net/browse/PB-69)

---

## Changes

Adds more feature flags on the page, to include:

random vs Facebook sorting
With vs without social metrics
Engagement based vs random vs something smart (content based) sorting

The engagement based could also be a third condition in experiment
with vs without labels

---

## Concerns

Concerns with flags sometimes not loading prior to project-berlin-extension page loading.
Things tried:
-setting the flags-store to be asynchronous
-wrapping the flags-store in a function and calling it in different +page and +layout files
-wrapping the switch statement in an onFeatureFlags - this solution doesn't work for components
-wrapping switch/flags in an if block to check that the Object is not empty - this solution will keep the page from breaking, but it will also make it so that reactions, time posted, and labels will not show if $flags obj is empty

---

## Things to consider 

In order to solve the problem of the flags sometimes loading after the page loads, we could consider server-side initialization and Posthog bootstrapping

---

## Requirements Before Merging

none

### CODE REVIEW

File: src/routes/project-berlin-experiment/+page.svelte [hunk 1]
```diff
@@ -18,6 +19,29 @@
...
+	const sortPosts = (category) => {
+		return data.posts.posts.sort((a, b) =>
+			a.category === category && b.category !== category ? -1 : 1
+		);
+	};
+
+	if ($flags.should_sort_random.enabled) {
+		data.posts.posts.sort(() => Math.random() - 0.5);
+	}
+	if ($flags.should_emphasize_organic_posts.enabled) {
```
Comment (skinofstars) [inline src/routes/project-berlin-experiment/+page.svelte:32]: thsi should be a switch!

**Review by skinofstars — CHANGES_REQUESTED** [review]
Is there a way in Posthog to limit these to only be able to select one?

**Comment by michellejanay** [comment]
> Is there a way in Posthog to limit these to only be able to select one?

As in only have one at a time enabled? We can set the conditions to match the studyID and also specify the percentage of people it gets rolled out to

Also I will update the PR Template and make it a switch statement!

**Review by skinofstars — APPROVED** [review]

File: src/routes/project-berlin-experiment/+page.svelte [hunk 2]
```diff
@@ -15,9 +16,39 @@
...
+		case $flags.should_emphasize_organic_posts.enabled:
+			sortPosts('ORGANIC');
+			break;
+		case $flags.should_emphasize_engagement_posts.enabled:
+			sortPosts('ENGAGEMENT');
+			break;
+		case $flags.should_emphasize_sponsored_posts.enabled:
+			sortPosts('SPONSORED');
+			break;
+		case $flags.should_not_show_sponsored.enabled:
```
Comment (skinofstars) [inline src/routes/project-berlin-experiment/+page.svelte:48]: you could remove sponsored and priolritise engagement

File: src/routes/project-berlin-experiment/+page.svelte [hunk 3]
```diff
@@ -15,9 +16,39 @@
...
 	const { prolific_pid, study_id, session_id, form_id } = data.prolificParams;
 
 	const offBoardLink = `${PUBLIC_TYPEFORM_LINK}/${form_id}#prolific_pid=${prolific_pid}&study_id=${study_id}&session_id=${session_id}&offboarding=${true}`;
+
+	switch (true) {
+		case $flags.should_sort_random.enable:
+			data.posts.posts.sort(() => Math.random() - 0.5);
+			break;
+		case $flags.should_emphasize_organic_posts.enabled:
+			sortPosts('ORGANIC');
```
Comment (skinofstars) [inline src/routes/project-berlin-experiment/+page.svelte:40]: let posts = data.posts.posts;

// order block
posts = sortBy(posts, 'ORGANIC')

// hide block
posts = removeBy(posts, 'SPONSORED')

File: src/routes/project-berlin-experiment/+page.svelte [hunk 4]
```diff
@@ -5,10 +5,43 @@
...
+					if (data.category === 'ORGANIC') {
+						return 1;
+					}
+				});
+				break;
+			case $flags.should_emphasize_engagement_posts.enabled:
+				data.posts.posts = _.sortBy(data.posts.posts, ['category']);
+				break;
+			case $flags.should_emphasize_sponsored_posts.enabled:
+				console.log('emphasize sponsored');
```
Comment (skinofstars) [inline src/routes/project-berlin-experiment/+page.svelte:35]: remove console

**Review by skinofstars — APPROVED** [review]
console to remove, but that's it really

**Pull request closed without merging**